File: demo/src/main/java/com/example/demo/controller/HelloController.java
package com.example.demo.controller;

import lombok.extern.slf4j.Slf4j;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/")
public class HelloController {
    private final static Logger logger = LoggerFactory.getLogger(HelloController.class);

    @RequestMapping("/hello/{name}")
    public String hello(@PathVariable(value = "name") final String name) {
        logger.debug("get name" + name);
        return "Hello " + name;
    }
}

File: demo-kotlin/src/main/kotlin/com/example/demo/controller/HelloController.kt
package com.example.demo.controller

import lombok.extern.slf4j.Slf4j
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Slf4j
@RestController
@RequestMapping("/")
class HelloController {
    private val logger = LoggerFactory.getLogger(HelloController::class.java)

    @RequestMapping("/hello/{name}")
    fun hello(@PathVariable(value = "name") name: String): String {
        logger.debug("get name is $name")
        return "Hello $name"
    }
}
